### ISSUE TITLE
Attributes such as `#[non_exhaustive]` are not considered

### ISSUE THREAD

**Issue by Enselic** [Enselic/public_items#21]
Adding e.g. `#[non_exhaustive]` to an enum can be a breaking API change, but attributes are currently not considered at all by `public_items`.

**Comment by douweschulte** [comment]
Attributes are saved in `Item::attrs` as a `Vec<String>` so it is possible to implement something like this. Of interest are at least:
* non_exhaustive
* cfg/doc_cfg
* must_use

This will likely have to wait until #34, to include this functionality.

**Comment by Enselic** [comment]
Yeah, I have actually been thinking if maybe we should change our public API structure from a list of PublicItems to a tree of PublicItems. I think that will allow us to solve #34 quite easily. And with a tree structure it would perhaps be more natural to annotate items in the tree with these kinds of attributes. These are just loose thoughts at the moment, but I think I might be on to something. I saw your updated PR btw, but will need some more time to take a look at it. I think it is ready for a deeper look now :)

**Comment by douweschulte** [comment]
I do wholeheartedly agree with the tree data structure. After the PR (which is indeed ready for a full review, I will still be cleaning it but no major work) I will be happy to discuss it more.
